include/chemfiles/misc.hpp

```diff
@@ -27,29 +27,27 @@ typedef std::function<void(const std::string& message)> warning_callback_t; // N
 /// @param callback callback function that will be called on each warning
 void CHFL_EXPORT set_warning_callback(warning_callback_t callback);
 
-/// Read configuration data from the file at `path`.
+/// Get the list of formats chemfiles knows about, and all associated metadata
 ///
-/// By default, chemfiles reads configuration from any file named
-/// `.chemfiles.toml` or `chemfiles.toml` in the current directory or any parent
-/// directory. This function can be used to add data from another configuration
-/// file.
+/// @example{formats_list.cpp}
+std::vector<std::reference_wrapper<const FormatMetadata>> CHFL_EXPORT formats_list();
+
+/// Get the format that chemfiles would use to read a file at the given
+/// path.
 ///
-/// This function will throw a `ConfigurationError` if there is no file at
-/// `path`, or if the file is incorrectly formatted. Data from the new
-/// configuration file will overwrite any existing data.
+/// The format is only guessed from the filename extension, chemfiles does not
+/// currently read the file to guess the format.
```

Comment:
Not opening also means that a successful guess does not mean that the format is actually supported (see discussion in Trajectory.cpp). Same wording also in the C API.